underscore.comparison.islike.js

```diff
@@ -0,0 +1,70 @@
+/*
+* Tests if an object is like another. This means objects should follow the same
+* structure and arrays should contain the same types.
+*
+* E.g.
+*
+* _.islike(
+*   {name: "James", age: 10, hobbies: ["football", "computer games", "baking"]},
+*   {name: "", age: 0, hobbies: [""]}
+* )
+*/
+(function() {
+  // Establish the root object, `window` in the browser, or `require` it on the server.
+  if (typeof exports === 'object') {
+    _ = module.exports = require('underscore');
+  }
+
+    var islike = function(obj, pattern) {
+        if (typeof pattern === "function") {
+            return obj instanceof pattern;
+        }
+
+        if (typeof obj !== typeof pattern) return false;
+            if (pattern instanceof Array && !(obj instanceof Array)) return false;
+
+                var type = typeof pattern;
+
+                if (type == "object") {
+                    if (pattern instanceof Array) {
+                        if (pattern.length > 0) {
+                            var oTypes = _.uniq(_.map(obj, fTypeof));
+                            var pTypes = _.uniq(_.map(pattern, fTypeof));
+                            if (_.difference(oTypes, pTypes).length) {
+                                return false;
+                            }
+                        }
+                    } else { // object
+                        if (pattern.constructor === pattern.constructor.prototype.constructor) {
+                            // for 'simple' objects we enumerate
+                            for (var k in pattern) {
+                                if (!pattern.hasOwnProperty(k)) {
```

Comment:
Sorry, the official name for `_.any` is _.some (any is an alias)

On Fri, Jun 19, 2015 at 11:20 AM, Joe Bain <notifications@github.com> wrote:

> In underscore.comparison.islike.js
> <https://github.com/documentcloud/underscore-contrib/pull/196#discussion_r32838824>
> :
>
> > +                var type = typeof pattern;
> > +
> > +                if (type == "object") {
> > +                    if (pattern instanceof Array) {
> > +                        if (pattern.length > 0) {
> > +                            var oTypes = _.uniq(_.map(obj, fTypeof));
> > +                            var pTypes = _.uniq(_.map(pattern, fTypeof));
> > +                            if (_.difference(oTypes, pTypes).length) {
> > +                                return false;
> > +                            }
> > +                        }
> > +                    } else { // object
> > +                        if (pattern.constructor === pattern.constructor.prototype.constructor) {
> > +                            // for 'simple' objects we enumerate
> > +                            for (var k in pattern) {
> > +                                if (!pattern.hasOwnProperty(k)) {
>
> Sorry, I can't find _.any in underscore or contrib. I think I'm being
> blind, where is that?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/documentcloud/underscore-contrib/pull/196/files#r32838824>
> .
>